abdiff/core/calc_ab_metrics.py

```diff
@@ -215,17 +217,16 @@ def _prepare_duckdb_context(
     return fields, sources
 
 
-def _get_global_counts(conn: DuckDBPyConnection, fields: list[str]) -> tuple[int, int]:
+def _get_global_counts(conn: DuckDBPyConnection) -> tuple[int, int]:
     total_records = conn.execute(
         """
         SELECT COUNT(*) FROM record_diff_matrix
         """
     ).fetchone()[0]
 
-    any_field_modified_condition = " OR ".join(f"{field} = 1" for field in fields)
     total_records_with_diff = conn.execute(
-        f"""
-        select count(*) from record_diff_matrix where {any_field_modified_condition}
+        """
+        select count(*) from record_diff_matrix where has_diff;
```

Comment:
Hmm, should this query read

```sql
select count(*) from record_diff_matrix where has_diff=1
```
based on the [derivation for `has_diff`](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/31/files#diff-9080a340ec2806075e422a9e58284e7dee2a0c7ccf5f4ff2d8178c03ec5e1c97R85)?

Reply:
Your right that `had_diff` is getting either the integer `1` or `0`, but you can treat this as a boolean when querying DuckDB (and most DBs).

e.g.

```

-- 1 is considered TRUE
D select 1 == true;
┌────────────────────────────┐
│ (1 = CAST('t' AS BOOLEAN)) │
│          boolean           │
├────────────────────────────┤
│ true                       │
└────────────────────────────┘
D select 0 == true;
┌────────────────────────────┐
│ (0 = CAST('t' AS BOOLEAN)) │
│          boolean           │
├────────────────────────────┤
│ false                      │
└────────────────────────────┘

-- 0 is considered FALSE
D select 0 == true;
┌────────────────────────────┐
│ (0 = CAST('t' AS BOOLEAN)) │
│          boolean           │
├────────────────────────────┤
│ false                      │
└────────────────────────────┘
D select 0 == false;
┌────────────────────────────┐
│ (0 = CAST('f' AS BOOLEAN)) │
│          boolean           │
├────────────────────────────┤
│ true                       │
└────────────────────────────┘
```

So, safe to just query like `where has_diff` which implies `where has_diff = 1`.

Reply:
Hmm, is the "casting to boolean" default behavior? 🤔

Reply:
@jonavellecuerdo - to my knowledge, yes it is.  I don't lean into this convention all that often, preferring the explicitness of true boolean values where possible (though.... they likely just store a `1` or `0` under the hood in the given system).  But in this sparse matrix, where conversions between python / pyarrow / parquet / duckdb could be subtly change types, a simple integer felt ideal.

The same is even true in python, further demonstrating it's kind of a "default" behavior in many systems:

```
In [1]: assert 1 == True

In [2]: assert 1 == False
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 assert 1 == False

AssertionError: 

In [3]: assert 0 == True
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[3], line 1
----> 1 assert 0 == True

AssertionError: 

In [4]: assert 0 == False
```